presto-iceberg/src/main/java/com/facebook/presto/iceberg/TypeConverter.java

```diff
@@ -0,0 +1,438 @@
+        }
+        if (type instanceof IntegerType) {
+            return Types.IntegerType.get();
+        }
+        if (type instanceof BigintType) {
+            return Types.LongType.get();
+        }
+        if (type instanceof RealType) {
+            return Types.FloatType.get();
+        }
+        if (type instanceof DoubleType) {
+            return Types.DoubleType.get();
+        }
+        if (type instanceof DecimalType) {
+            return fromDecimal((DecimalType) type);
+        }
+        if (type instanceof VarcharType) {
+            return Types.StringType.get();
+        }
+        if (type instanceof VarbinaryType) {
+            return Types.BinaryType.get();
+        }
+        if (type instanceof DateType) {
+            return Types.DateType.get();
+        }
+        if (type instanceof RowType) {
+            return fromRow((RowType) type);
+        }
+        if (type instanceof ArrayType) {
+            return fromArray((ArrayType) type);
+        }
+        if (type instanceof MapType) {
+            return fromMap((MapType) type);
+        }
+        if (type instanceof TimeType) {
+            throw new PrestoException(NOT_SUPPORTED, format("Time not supported for Iceberg."));
+        }
+        if (type instanceof TimestampType) {
+            throw new PrestoException(NOT_SUPPORTED, format("Timestamp not supported for Iceberg."));
+        }
+        if (type instanceof TimestampWithTimeZoneType) {
+            throw new PrestoException(NOT_SUPPORTED, format("Timestamp with timezone not supported for Iceberg."));
+        }
+        throw new PrestoException(NOT_SUPPORTED, "Type not supported for Iceberg: " + type.getDisplayName());
+    }
+
+    public static HiveType toHiveType(Type type)
+    {
+        return HiveType.toHiveType(toHiveTypeInfo(type));
+    }
+
+    private static org.apache.iceberg.types.Type fromDecimal(DecimalType type)
+    {
+        return Types.DecimalType.of(type.getPrecision(), type.getScale());
+    }
+
+    private static org.apache.iceberg.types.Type fromRow(RowType type)
+    {
+        List<Types.NestedField> fields = new ArrayList<>();
+        for (RowType.Field field : type.getFields()) {
+            String name = field.getName().orElseThrow(() ->
+                    new PrestoException(NOT_SUPPORTED, "Row type field does not have a name: " + type.getDisplayName()));
+            fields.add(Types.NestedField.optional(fields.size() + 1, name, toIcebergType(field.getType())));
+        }
+        return Types.StructType.of(fields);
+    }
+
+    private static org.apache.iceberg.types.Type fromArray(ArrayType type)
+    {
+        return Types.ListType.ofOptional(1, toIcebergType(type.getElementType()));
+    }
+
+    private static org.apache.iceberg.types.Type fromMap(MapType type)
+    {
+        return Types.MapType.ofOptional(1, 2, toIcebergType(type.getKeyType()), toIcebergType(type.getValueType()));
+    }
+
+    private static TypeInfo toHiveTypeInfo(Type type)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 243-252, lines 253-261, lines 288-305 into separate methods.